### DISCUSSION TITLE
Adding selectable title to value and button controls?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [godotengine/godot-proposals#10065] [GUI]
What do you guys think of adding a title widget to more controls similar to the way it is for CheckBox and CheckButton?

For example Buttons, Sliders, OptionButtons and SpinBoxes could all receive a title section that could be configured to either left, right or none, so it can look like these...

`MyAction PressMe`
`PressMe MyAction` 
`PressMe` 

or for something like a spinbox...
`Hp 0 ^v`
`0 ^v Hp`
`0 ^v`

with `^v` being the up and down icons for the spin slider.

If OptionButton and SpinBox is ever extended to support more gamepad friendly menus like...
`Hp < 0 >`
`< 0 > Hp`
`< 0 >`

This feature would be nice since it would essentially allow the title to be able to copy what state the value control is in.
So if lets say if it where a button, the Button if focused, hovered, pressed, or disabled the title would be too.

Right now there is not way to share state with a control, so right now we would have to export our own theme properties and manually edit the title's theme when the value's state changes.  

I think this would just be better

**Comment by Calinou** [comment]
Displaying a name for an OptionButton or Slider is typically done by nesting them within a HBoxContainer, putting a Label with the name, and putting another Label at the end to display the slider's value.

There might be value into combining this functionality in a single node, but this will also limit flexibility for users to theme things as they wish. We can only make each node so complex as to avoid making its inspector unusable.

**Reply by Shadowblitz16** [reply]
The main point is that control state is not shared so composite controls don't act as one control.
